File: lua-5.3.5/Student.cpp
#include "Student.h"

Student::Student() :_name("Empty"), _age(0) {
	std::cout << "Student Constructor" << std::endl;
}

Student::~Student() {
	std::cout << "Student Destructor" << std::endl;
}

std::string Student::get_name()
{
	return _name;
}

void Student::set_name(std::string name)
{
	_name = name;
}

unsigned Student::get_age()
{
	return _age;
}

void Student::set_age(unsigned age)
{
	_age = age;
}

void Student::print()
{
	std::cout << "name:" << _name << " age:" << _age << std::endl;
}

File: lua-5.3.5/student.lua
--local student_obj = Student.create()
--Student.set_name(student_obj,"Jack")
--Student.set_age(student_obj,20)
--print(Student.get_name(student_obj))
--print(Student.get_age(student_obj))
--Student.print(student_obj)



local student_obj = Student.create()
student_obj:set_name("Jack")
student_obj:set_age(20)
print(student_obj:get_name())
print(student_obj:get_age())
student_obj:print()
print(student_obj)
student_obj = nil
collectgarbage("collect")